### ISSUE TITLE
shap_interaction_values

### ISSUE THREAD

**Issue by mdjabc** [shap/shap#1438]
WHen I use shap_interaction_values for catboost, some  problem:
1. 'TreeEnsemble' object has no attribute 'values'.
2.  the calculated interaction_values are Nan or 0.
When I use shap for xgboost , the question  2 also is existed.

**Comment by mdjabc** [comment]
explainer = shap.TreeExplainer(model)
shap_values = explainer.shap_values(test_x)
shap_interaction_values = shap.TreeExplainer(model).shap_interaction_values(test_x.iloc[:50,:])

**Comment by mdjabc** [comment]
But it can work
prediction_diff = model.get_feature_importance(type='ShapInteractionValues', data=eval_dataset, prettified=True)

**Comment by slundberg** [comment]
Can you share a complete example? This works for me on XGB:

```
import shap
import xgboost

# get a dataset on income prediction
X,y = shap.datasets.adult()

# train an XGBoost model (but any other model type would also work)
model = xgboost.XGBClassifier()
model.fit(X, y)

explainer = shap.Explainer(model)
explainer.shap_interaction_values(X)
```

**Comment by mdjabc** [comment]
catboost==0.23.0
shap==0.34.0/0.35.0
from catboost import CatBoostClassifier,Pool
import shap
from sklearn import datasets
data = datasets.load_breast_cancer()
X = data.data
y = data.target
eval_dataset = Pool(data=X,
                    label=y,
                    )
model=CatBoostClassifier()#,verbose=False
model.fit(X, y)
explainer = shap.TreeExplainer(model)
shap_interaction_values = shap.TreeExplainer(model).shap_interaction_values(X)
print(shap_interaction_values)
the shap_interaction_values are Nan or 0.
However ,I can use the code(prediction_diff = model.get_feature_importance(type='ShapInteractionValues', data=eval_dataset, prettified=True)) to calculate feature_importance, indicating the shap_interaction_valuescan be calculated.

**Comment by hal-314** [comment]
I got ` 'TreeEnsemble' object has no attribute 'values'.` when I train a LighGBM regressor/classifier with categorical features. 

To reproduce:
```
import shap
import lightgbm as lgb

# get a dataset on income prediction
X,y = shap.datasets.adult()

# train an XGBoost model (but any other model type would also work)
model = lgb.LGBMClassifier()
model.fit(X, y, categorical_feature=['Workclass','Marital Status']) # If categorical_feature=[] , shap_interaction_values() works 

explainer = shap.TreeExplainer(model)
explainer.shap_values(X) # <- it always works 
explainer.shap_interaction_values(X)
```
I have:
lightgbm==3.0.0
shap==0.36.0

**Comment by mayer79** [comment]
@hal-314 Same issue here https://github.com/slundberg/shap/issues/1644 with slightly newer package versions. Did you find a solution for this?

**Comment by hal-314** [comment]
No, I couldn't. At the end, I didn't use it.

**Comment by ThomasWolf0701** [comment]
Same problem for me interactions do not work with catboost:

bestModelShap = explainer.shap_interaction_values( transformedMatrix.values())
Traceback (most recent call last):

  File "<ipython-input-205-deb27fe10bd5>", line 1, in <module>
    bestModelShap = explainer.shap_interaction_values( transformedMatrix.values())

TypeError: 'numpy.ndarray' object is not callable


bestModelShap = explainer.shap_interaction_values( transformedMatrix)
Traceback (most recent call last):

  File "<ipython-input-206-f277cad93005>", line 1, in <module>
    bestModelShap = explainer.shap_interaction_values( transformedMatrix)

  File "C:\Users\Thomas Wolf\anaconda3\envs\ChemPipeline\lib\site-packages\shap\explainers\_tree.py", line 487, in shap_interaction_values
    X, y, X_missing, flat_output, tree_limit, _ = self._validate_inputs(X, y, tree_limit, False)

  File "C:\Users\Thomas Wolf\anaconda3\envs\ChemPipeline\lib\site-packages\shap\explainers\_tree.py", line 235, in _validate_inputs
    tree_limit = self.model.values.shape[0]

AttributeError: 'TreeEnsemble' object has no attribute 'values'

**Comment by whz789** [comment]
Catboost can not use "shap_interaction_values"! AttributeError: 'TreeEnsemble' object has no attribute 'values'

**Comment by QiaomeiFeng** [comment]
> Catboost can not use "shap_interaction_values"! AttributeError: 'TreeEnsemble' object has no attribute 'values'

So what kind of algorithm can use "shap_interaction_values"?  I met AttributeError: 'Linear' object has no attribute 'shap_interaction_values'. Thanks for any help.

**Comment by nilslacroix** [comment]
Catboost can not use shap_values either. Same error.

**Comment by CloseChoice** [comment]
closed via #3457
